tests/macros/DefaultAPISidebar.test.js

```diff
@@ -0,0 +1,359 @@
+* are locale-sensitive sometimes, and sometimes include a title
+* attribute which we have to check.
+*/
+function checkGuideItem(expected, actual, locale) {
+    // Check the textContent
+    const expectedItem = expectedGuides[locale][expected];
+    expect(actual.textContent).toEqual(expectedItem.text);
+    // Check the target
+    const itemLink = actual.querySelector('a');
+    expect(itemLink.href).toEqual(expectedItem.target);
+    // Check the title
+    if (expectedItem.title) {
+        expect(itemLink.getAttribute('title')).toEqual(expectedItem.title);
+    }
+}
+
+/**
+* Check a subsection of the sidebar, represented as a <details> element
+* containing a <summary> and a list of links.
+*/
+function checkSubList(name, config, details, checker, next) {
+    const hasSubList = config.expected.details[name];
+    if (hasSubList) {
+        // Check the summary
+        const expectedSummary = commonL10nJSON[name][config.locale];
+        const actual = details[next];
+        const actualSummary = actual.querySelector('summary');
+        expect(actualSummary.textContent).toEqual(expectedSummary);
+        // Check the list
+        const actualItems = actual.querySelectorAll('ol>li');
+        const expectedItems = config.expected.details[name];
+        expect(actualItems.length).toEqual(expectedItems.length);
+        for (let i = 0; i < actualItems.length; i++) {
+            checker(expectedItems[i], actualItems[i], config.locale);
+        }
+        next++;
+    }
+    return next;
+}
+
+/**
+* This is the entry point for checking the result of a test.
+* config.expected contains the expected results, and we use other bits
+* of config, most notably locale.
+*/
+function checkResult(html, config) {
+    // Lint the HTML
+    expect(lintHTML(html)).toBeFalsy();
+
+    const dom = JSDOM.fragment(html);
+    // Check that all links reference the proper locale or use https
+    const num_total_links = dom.querySelectorAll('a[href]').length;
+    const num_valid_links = dom.querySelectorAll(`a[href^="/${config.locale}/"], a[href^="https://"]`).length;
+    expect(num_valid_links).toEqual(num_total_links);
+
+    if (config.expected.overview) {
+        // Test overview link
+        const overviewLink = dom.querySelector('ol>li>strong');
+        checkItem(config.expected.overview, overviewLink, config.locale);
+    }
+
+    // Test sublists
+    const details = dom.querySelectorAll('ol>li>details');
+    expect(details.length).toEqual(Object.keys(config.expected.details).length);
+    // "_" prefix is to suppress lint's no-unused-vars
+    let _next = 0;
+    _next = checkSubList('Guides', config, details, checkGuideItem, _next);
+    _next = checkSubList('Interfaces', config, details, checkItem, _next);
+    _next = checkSubList('Properties', config, details, checkItem, _next);
+    _next = checkSubList('Methods', config, details, checkItem, _next);
+    _next = checkSubList('Events', config, details, checkItem, _next);
```

Comment:
Super nit: The `_` could be removed if this last variable assignment wasn't done, so this would work as well:
```
let next = 0;
next = checkSubList('Guides', config, details, checkGuideItem, next);
next = checkSubList('Interfaces', config, details, checkItem, next);
next = checkSubList('Properties', config, details, checkItem, next);
next = checkSubList('Methods', config, details, checkItem, next);
checkSubList('Events', config, details, checkItem, next);
```
but that's a personal style preference.